### ISSUE TITLE
Unhandled issue during upgrade to 4.4.0

### ISSUE THREAD

**Issue by MiguelCasaresRobles** [wazuh/wazuh#16889]
|Wazuh version|Component|Install type|Install method|Platform|
|---|---|---|---|---|
| 4.4.0 | Upgrade | Manager| Package | amd64 |

# Description 
Hello Team,
Upgrading to Wazuh 4.4.0 for a reason I do not know, the upgrade failed, and I tried again and the upgrade finished without apparent problems.
Here the failed installation:
```shell
# apt-get install wazuh-manager
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following package was automatically installed and is no longer required:
  linux-image-5.10.0-13-amd64
Use 'apt autoremove' to remove it.
Suggested packages:
  expect
The following packages will be upgraded:
  wazuh-manager
1 upgraded, 0 newly installed, 0 to remove and 52 not upgraded.
Need to get 122 MB of archives.
After this operation, 7,299 kB of additional disk space will be used.
Get:1 https://packages.wazuh.com/4.x/apt stable/main amd64 wazuh-manager amd64 4.4.0-1 [122 MB]
Fetched 122 MB in 4s (29.6 MB/s)        
Reading changelogs... Done
(Reading database ... 65239 files and directories currently installed.)
Preparing to unpack .../wazuh-manager_4.4.0-1_amd64.deb ...
dpkg: error processing archive /var/cache/apt/archives/wazuh-manager_4.4.0-1_amd64.deb (--unpack):
 new wazuh-manager package pre-installation script subprocess returned error exit status 1
Errors were encountered while processing:
 /var/cache/apt/archives/wazuh-manager_4.4.0-1_amd64.deb
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

Here the correct installation:
```shell
# apt-get install wazuh-manager
Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
The following package was automatically installed and is no longer required:
  linux-image-5.10.0-13-amd64
Use 'apt autoremove' to remove it.
Suggested packages:
  expect
The following packages will be upgraded:
  wazuh-manager
1 upgraded, 0 newly installed, 0 to remove and 52 not upgraded.
Need to get 0 B/122 MB of archives.
After this operation, 7,299 kB of additional disk space will be used.
Reading changelogs... Done
(Reading database ... 65239 files and directories currently installed.)
Preparing to unpack .../wazuh-manager_4.4.0-1_amd64.deb ...
Unpacking wazuh-manager (4.4.0-1) over (4.3.9-1) ...
Setting up wazuh-manager (4.4.0-1) ...
```

Then I evidenced some data is missing from the Wazuh Manager, like RBAC configurations. I had some role mappings, and when I list them, I see only the default mappings:
```json
{
   "data": {
      "affected_items": [
         {
            "id": 1,
            "name": "wui_elastic_admin",
            "rule": {
               "FIND": {
                  "username": "elastic"
               }
            },
            "roles": [
               1
            ]
         },
         {
            "id": 2,
            "name": "wui_opendistro_admin",
            "rule": {
               "FIND": {
                  "user_name": "admin"
               }
            },
            "roles": [
               1
            ]
         }
      ],
      "total_affected_items": 2,
      "total_failed_items": 0,
      "failed_items": []
   },
   "message": "All specified security rules were returned",
   "error": 0
}
```

Then checking more, I see other data is missing, that led me to check the installation directory and I can evidence, there are backups of configuration files, that were not restored:

<details>
  <summary>list of .save files</summary>

```shell
# find /var/ossec/ -type f | grep -E ".save$"
/var/ossec/api/configuration/ssl/server.crt.save
/var/ossec/api/configuration/ssl/server.key.save
/var/ossec/api/configuration/api.yaml.save
/var/ossec/api/configuration/security/private_key.pem.save
/var/ossec/api/configuration/security/public_key.pem.save
/var/ossec/api/scripts/wazuh-apid.py.save
/var/ossec/etc/decoders/cybereason-decoders.xml.save
/var/ossec/etc/decoders/local_decoder.xml.save
/var/ossec/etc/decoders/radware_decoders.xml.save
/var/ossec/etc/decoders/eset-bsd-decoders.xml.save
/var/ossec/etc/decoders/eset-decoders.xml.save
/var/ossec/etc/decoders/sysmon_decoder.xml.save
/var/ossec/etc/decoders/kaspersky-decoders.xml.save
/var/ossec/etc/decoders/CEF-decoder-Manitou.xml.save
/var/ossec/etc/decoders/cisco-asa-decoders.xml.save
/var/ossec/etc/decoders/esxi-nico-decoders.xml.save
/var/ossec/etc/decoders/cisco-ios-decoders.xml.save
/var/ossec/etc/decoders/pfsense-custom-decoders.xml.save
/var/ossec/etc/decoders/imperva-decoders.xml.save
/var/ossec/etc/decoders/weblogic-decoders.xml.save
/var/ossec/etc/decoders/meraki-ap-decoder.xml.save
/var/ossec/etc/decoders/veritas-decoders.xml.save
/var/ossec/etc/decoders/forcepoint-dlp-decoders.xml.save
/var/ossec/etc/decoders/cybereason-syslog-decoders.xml.save
/var/ossec/etc/decoders/fortimail-decoders.xml.save
/var/ossec/etc/decoders/CEF-varios.xml.save
/var/ossec/etc/decoders/aruba_decoders.xml.disabled.save
/var/ossec/etc/decoders/fortiddos-decoders.xml.save
/var/ossec/etc/decoders/meraki-firewall-decoder.xml.save
/var/ossec/etc/decoders/fortigate-decoders.xml.save
/var/ossec/etc/decoders/traefik-decoders.xml.save
/var/ossec/etc/decoders/apache-new.xml.save
/var/ossec/etc/decoders/sql-siblings.xml.save
/var/ossec/etc/decoders/karspersky-custom-decoders.xml.save
/var/ossec/etc/localtime.save
/var/ossec/etc/rootcheck/rootkit_files.txt.save
/var/ossec/etc/rootcheck/win_audit_rcl.txt.save
/var/ossec/etc/rootcheck/cis_mysql5-6_enterprise_rcl.txt.save
/var/ossec/etc/rootcheck/cis_debian_linux_rcl.txt.save
/var/ossec/etc/rootcheck/system_audit_ssh.txt.save
/var/ossec/etc/rootcheck/cis_rhel6_linux_rcl.txt.save
/var/ossec/etc/rootcheck/cis_mysql5-6_community_rcl.txt.save
/var/ossec/etc/rootcheck/win_applications_rcl.txt.save
/var/ossec/etc/rootcheck/system_audit_rcl.txt.save
/var/ossec/etc/rootcheck/win_malware_rcl.txt.save
/var/ossec/etc/rootcheck/cis_rhel5_linux_rcl.txt.save
/var/ossec/etc/rootcheck/cis_win2012r2_domainL2_rcl.txt.save
/var/ossec/etc/rootcheck/cis_apache2224_rcl.txt.save
/var/ossec/etc/rootcheck/cis_win2012r2_domainL1_rcl.txt.save
/var/ossec/etc/rootcheck/cis_rhel_linux_rcl.txt.save
/var/ossec/etc/rootcheck/cis_win2012r2_memberL1_rcl.txt.save
/var/ossec/etc/rootcheck/cis_rhel7_linux_rcl.txt.save
/var/ossec/etc/rootcheck/cis_win2012r2_memberL2_rcl.txt.save
/var/ossec/etc/rootcheck/cis_sles12_linux_rcl.txt.save
/var/ossec/etc/rootcheck/cis_sles11_linux_rcl.txt.save
/var/ossec/etc/rootcheck/rootkit_trojans.txt.save
/var/ossec/etc/sslmanager.cert.save
/var/ossec/etc/local_internal_options.conf.save
/var/ossec/etc/rules/jumpcloud-rules.xml.save
/var/ossec/etc/rules/aruba_rules.xml.disabled.save
/var/ossec/etc/rules/malwareBazaar_Rules.xml.save
/var/ossec/etc/rules/fortimail-rules.xml.save
/var/ossec/etc/rules/local_rules.xml.save
/var/ossec/etc/rules/cisco-asa-rules.xml.save
/var/ossec/etc/rules/windows-custom.xml.save
/var/ossec/etc/rules/cybereason-rules.xml.save
/var/ossec/etc/rules/attack_rules.xml.save
/var/ossec/etc/rules/active-response-nmap.xml.save
/var/ossec/etc/rules/kaspersky-rules.xml.save
/var/ossec/etc/rules/pfsense-custom-rules.xml.save
/var/ossec/etc/rules/eset-rules.xml.save
/var/ossec/etc/rules/meraki-rules.xml.save
/var/ossec/etc/rules/fortigate-rules.xml.save
/var/ossec/etc/rules/fortiddos-rules.xml.save
/var/ossec/etc/rules/Domain_Route53.xml.save
/var/ossec/etc/rules/windows-auditing-rules.xml.save
/var/ossec/etc/rules/cluster-monitor-rules.xml.save
/var/ossec/etc/rules/weblogic-rules.xml.save
/var/ossec/etc/ossec.conf.new.save
/var/ossec/etc/ossec.conf.save
/var/ossec/etc/client.keys.save
/var/ossec/etc/sslmanager.key.save
/var/ossec/etc/internal_options.conf.save
/var/ossec/etc/github-ent-last-id.save
/var/ossec/etc/lists/security-eventchannel.cdb.save
/var/ossec/etc/lists/darktrace.save
/var/ossec/etc/lists/amazon/aws-eventnames.save
/var/ossec/etc/lists/amazon/aws-eventnames.cdb.save
/var/ossec/etc/lists/amazon/aws-sources.save
/var/ossec/etc/lists/security-eventchannel.save
/var/ossec/etc/lists/win-privaccounts.cdb.save
/var/ossec/etc/lists/tor-list.cdb.save
/var/ossec/etc/lists/tor-list.save
/var/ossec/etc/lists/forbidden-files.cdb.save
/var/ossec/etc/lists/win-privaccounts.save
/var/ossec/etc/lists/filter2.save
/var/ossec/etc/lists/report-weekly.save
/var/ossec/etc/lists/report-list.save
/var/ossec/etc/lists/test-sync.cdb.save
/var/ossec/etc/lists/forbidden-files.save
/var/ossec/etc/lists/testing.save
/var/ossec/etc/lists/disks-list.save
/var/ossec/etc/lists/audit-keys.save
/var/ossec/etc/lists/test-sync.save
/var/ossec/etc/lists/cdblist.save
/var/ossec/etc/lists/filter1.save
/var/ossec/etc/lists/audit-keys.cdb.save
/var/ossec/etc/shared/linux/merged.mg.save
/var/ossec/etc/shared/linux/agent.conf.save
/var/ossec/etc/shared/ar.conf.save
/var/ossec/etc/shared/default/wpk_root.pem.save
/var/ossec/etc/shared/default/rootkit_files.txt.save
/var/ossec/etc/shared/default/win_audit_rcl.txt.save
/var/ossec/etc/shared/default/cis_mysql5-6_enterprise_rcl.txt.save
/var/ossec/etc/shared/default/cis_debian_linux_rcl.txt.save
/var/ossec/etc/shared/default/system_audit_ssh.txt.save
/var/ossec/etc/shared/default/cis_rhel6_linux_rcl.txt.save
/var/ossec/etc/shared/default/cis_mysql5-6_community_rcl.txt.save
/var/ossec/etc/shared/default/win_applications_rcl.txt.save
/var/ossec/etc/shared/default/system_audit_rcl.txt.save
/var/ossec/etc/shared/default/win_malware_rcl.txt.save
/var/ossec/etc/shared/default/cis_rhel5_linux_rcl.txt.save
/var/ossec/etc/shared/default/cis_win2012r2_domainL2_rcl.txt.save
/var/ossec/etc/shared/default/cis_apache2224_rcl.txt.save
/var/ossec/etc/shared/default/merged.mg.save
/var/ossec/etc/shared/default/agent.conf.save
/var/ossec/etc/shared/default/cis_win2012r2_domainL1_rcl.txt.save
/var/ossec/etc/shared/default/cis_rhel_linux_rcl.txt.save
/var/ossec/etc/shared/default/cis_win2012r2_memberL1_rcl.txt.save
/var/ossec/etc/shared/default/cis_rhel7_linux_rcl.txt.save
/var/ossec/etc/shared/default/cis_win2012r2_memberL2_rcl.txt.save
/var/ossec/etc/shared/default/cis_sles12_linux_rcl.txt.save
/var/ossec/etc/shared/default/cis_sles11_linux_rcl.txt.save
/var/ossec/etc/shared/default/testing_bat.bat.save
/var/ossec/etc/shared/default/rootkit_trojans.txt.save
/var/ossec/etc/shared/default/update_wpk_ca.sh.save
/var/ossec/etc/shared/test-grp/merged.mg.save
/var/ossec/etc/shared/test-grp/agent.conf.save
/var/ossec/etc/shared/agent-template.conf.save
/var/ossec/etc/shared/winwks/svb_windows_Pre-PROD_all.yml.save
/var/ossec/etc/shared/winwks/merged.mg.save
/var/ossec/etc/shared/winwks/agent.conf.save
/var/ossec/etc/shared/test-wodle/merged.mg.save
/var/ossec/etc/shared/test-wodle/agent.conf.save
/var/ossec/etc/shared/winsrv/cis_sqlserver_2016.yml.save
/var/ossec/etc/shared/winsrv/cis_sqlserver_2019.yml.save
/var/ossec/etc/shared/winsrv/merged.mg.save
/var/ossec/etc/shared/winsrv/agent.conf.save
/var/ossec/etc/shared/winsrv/cis_sqlserver_2019_dev.yml.save
/var/ossec/var/multigroups/96f5f311/wpk_root.pem.save
/var/ossec/var/multigroups/96f5f311/rootkit_files.txt.save
/var/ossec/var/multigroups/96f5f311/win_audit_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_mysql5-6_enterprise_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/svb_windows_Pre-PROD_all.yml.save
/var/ossec/var/multigroups/96f5f311/cis_debian_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/system_audit_ssh.txt.save
/var/ossec/var/multigroups/96f5f311/cis_rhel6_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_mysql5-6_community_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/win_applications_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/system_audit_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/win_malware_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_rhel5_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_win2012r2_domainL2_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_apache2224_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/agent.conf.save
/var/ossec/var/multigroups/96f5f311/cis_win2012r2_domainL1_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_rhel_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_win2012r2_memberL1_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_rhel7_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_win2012r2_memberL2_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_sles12_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/cis_sles11_linux_rcl.txt.save
/var/ossec/var/multigroups/96f5f311/testing_bat.bat.save
/var/ossec/var/multigroups/96f5f311/rootkit_trojans.txt.save
/var/ossec/var/multigroups/96f5f311/update_wpk_ca.sh.save
/var/ossec/var/multigroups/fe375984/wpk_root.pem.save
/var/ossec/var/multigroups/fe375984/rootkit_files.txt.save
/var/ossec/var/multigroups/fe375984/win_audit_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_mysql5-6_enterprise_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_debian_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/system_audit_ssh.txt.save
/var/ossec/var/multigroups/fe375984/cis_rhel6_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_mysql5-6_community_rcl.txt.save
/var/ossec/var/multigroups/fe375984/win_applications_rcl.txt.save
/var/ossec/var/multigroups/fe375984/system_audit_rcl.txt.save
/var/ossec/var/multigroups/fe375984/win_malware_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_rhel5_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_win2012r2_domainL2_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_apache2224_rcl.txt.save
/var/ossec/var/multigroups/fe375984/agent.conf.save
/var/ossec/var/multigroups/fe375984/cis_win2012r2_domainL1_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_rhel_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_win2012r2_memberL1_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_rhel7_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_win2012r2_memberL2_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_sles12_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/cis_sles11_linux_rcl.txt.save
/var/ossec/var/multigroups/fe375984/testing_bat.bat.save
/var/ossec/var/multigroups/fe375984/rootkit_trojans.txt.save
/var/ossec/var/multigroups/fe375984/update_wpk_ca.sh.save
```

</details>

In conclusion, the issue is that when an upgrade fails, it should restore the state of the product (files and directories) to the state previous to the upgrade.
